lib/benchmark/bench.go

```diff
@@ -198,20 +202,20 @@ func (b *benchmarkThread) execute(measure *benchMeasure) {
 	<-done
 }
 
-func (b *benchmarkThread) sendMeasure(measure *benchMeasure) {
+func (b *benchmarkThread) sendMeasure(measure benchMeasure) {
```

Comment:
Instead of a generic `sendMeasure`, I'd use one for sending error values:
```go
func (b *benchmarkThread) sendErrorMeasure(err error, m benchMeasure) {
	m.End = time.Now()
	m.Error = err
	b.sendMeasure(m)
}
```
and a `sendCompletedMeasure` to send a value with `ThreadCompleted = true`. Btw, where is the `ThreadCompleted` used?

Reply:
`run()` in `bench.go`

```
func (b *benchmarkThread) run() {
	defer func() {
		if r := recover(); r != nil {
			logrus.Warningf("recover from panic: %v", r)
			b.sendMeasure(benchMeasure{ThreadCompleted: true})
		}
	}()

	for {
		select {
		case measure := <-b.receiveC:
			b.execute(measure)
		case <-b.ctx.Done():
			b.sendMeasure(benchMeasure{
				ThreadCompleted: true,
			})
			return
		}
	}
}

```

Reply:
A lot of the bench code is actually fixed here as there was a coordinated omission bug, https://github.com/gravitational/teleport/pull/4643